browser/module.go

```diff
@@ -69,3 +76,459 @@ func (*RootModule) NewModuleInstance(vu k6modules.VU) k6modules.Instance {
 func (mi *ModuleInstance) Exports() k6modules.Exports {
 	return k6modules.Exports{Default: mi.mod}
 }
+
+type mapping map[string]any
+
+// mapBrowserToGoja maps the browser API to the JS module.
+// The motivation of this mapping was to support $ and $$ wildcard
+// methods.
+// See issue #661 for more details.
+func mapBrowserToGoja(vu k6modules.VU) *goja.Object {
+	rt := vu.Runtime()
+
+	var (
+		obj         = rt.NewObject()
+		browserType = chromium.NewBrowserType(vu)
+	)
+	for k, v := range mapBrowserType(rt, browserType) {
+		err := obj.Set(k, rt.ToValue(v))
+		if err != nil {
+			k6common.Throw(rt, fmt.Errorf("mapping: %w", err))
+		}
+	}
+
+	return obj
+}
+
+func mapRequest(rt *goja.Runtime, r api.Request) mapping {
+	maps := mapping{
+		"allHeaders": r.AllHeaders,
+		"failure":    r.Failure,
+		"frame": func() *goja.Object {
+			mf := mapFrame(rt, r.Frame())
+			return rt.ToValue(mf).ToObject(rt)
+		},
+		"headerValue":         r.HeaderValue,
+		"headers":             r.Headers,
+		"headersArray":        r.HeadersArray,
+		"isNavigationRequest": r.IsNavigationRequest,
+		"method":              r.Method,
+		"postData":            r.PostData,
+		"postDataBuffer":      r.PostDataBuffer,
+		"postDataJSON":        r.PostDataJSON,
+		"redirectedFrom": func() *goja.Object {
+			mr := mapRequest(rt, r.RedirectedFrom())
+			return rt.ToValue(mr).ToObject(rt)
+		},
+		"redirectedTo": func() *goja.Object {
+			mr := mapRequest(rt, r.RedirectedTo())
+			return rt.ToValue(mr).ToObject(rt)
+		},
+		"resourceType": r.ResourceType,
+		"response": func() *goja.Object {
+			mr := mapResponse(rt, r.Response())
+			return rt.ToValue(mr).ToObject(rt)
+		},
+		"size":   r.Size,
+		"timing": r.Timing,
+		"uRL":    r.URL,
```

Comment:
is this really the name of the property `uRL`?

Reply:
Good catch. Resolved in https://github.com/grafana/xk6-browser/pull/673/commits/6cf7bcdb6792cd830a5ee2534d334c9a1e58357e.